extensions-core/s3-extensions/src/test/java/org/apache/druid/storage/s3/output/RetriableS3OutputStreamTest.java

```diff
@@ -0,0 +1,382 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.storage.s3.output;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.SdkClientException;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.model.AbortMultipartUploadRequest;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadRequest;
+import com.amazonaws.services.s3.model.CompleteMultipartUploadResult;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadRequest;
+import com.amazonaws.services.s3.model.InitiateMultipartUploadResult;
+import com.amazonaws.services.s3.model.PartETag;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+import com.amazonaws.services.s3.model.UploadPartResult;
+import org.apache.druid.java.util.common.IOE;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.storage.s3.NoopServerSideEncryption;
+import org.apache.druid.storage.s3.ServerSideEncryptingAmazonS3;
+import org.easymock.EasyMock;
+import org.hamcrest.CoreMatchers;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RetriableS3OutputStreamTest
+{
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  private final TestAmazonS3 s3 = new TestAmazonS3(0);
+  private final String path = "resultId";
+
+
+  private S3OutputConfig config;
+  private long maxResultsSize;
+  private long chunkSize;
+
+  @Before
+  public void setup() throws IOException
+  {
+    final File tempDir = temporaryFolder.newFolder();
+    chunkSize = 10L;
+    config = new S3OutputConfig()
+    {
+      @Override
+      public File getTempDir()
+      {
+        return tempDir;
+      }
+
+      @Override
+      public Long getChunkSize()
+      {
+        return chunkSize;
+      }
+
+      @Override
+      public long getMaxResultsSize()
+      {
+        return maxResultsSize;
+      }
+
+      @Override
+      public int getMaxTriesOnTransientError()
+      {
+        return 2;
+      }
+    };
+  }
+
+  @Test
+  public void testTooSmallChunkSize() throws IOException
+  {
+    maxResultsSize = 100000000000L;
```

Comment:
Help out our poor eyes? Insert underscores as the thousands separator? Or, since this number is really big, cast from a scientific notation representation?